src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,29 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        List<Integer> oddIndexNumbers = IntStream.range(0, numbers.size())
+                .mapToObj(i -> {
+                    int num = numbers.get(i);
+                    if (i % MOD_NUM_CHECK != ZERO_NUM) {
```

Comment:
```suggestion
                    if (i % MOD_NUM_CHECK != ZERO_NUM) {
```
"i % MOD_NUM_CHECK" extract it to method and reuse